src/lib/simulator/physics/Thermostat.h

```diff
@@ -32,10 +32,14 @@ class Thermostat {
   Thermostat() = default;
 
   /**
-     * @brief Applies the thermostat to adjust particle velocities.
-     * @param particles Container of particles to apply the thermostat to.
-     */
-  void inline apply(container::ParticleContainer<DIMENSIONS> &particles) {
+   * @brief Applies the thermostat to adjust particle velocities.
+   *        This method adjusts velocities by ignoring the total velocity of the fluid.
+   * @param particles Container of particles to apply the thermostat to.
+   */
+  void inline apply(container::Container<DIMENSIONS> &particles) {
+    auto average_velocity = calculateAverageVelocity(particles);
+    adjustVelocities(particles, average_velocity);
```

Comment:
I think you are including the wall particles here which leads to skewed results. The wall particles should not be considered as 'actual particles' but rather our way to model a wall